WordPress/Classes/Services/PostService+UnattachedMedia.swift

```diff
@@ -0,0 +1,22 @@
+import Foundation
+
+extension PostService {
+    @objc func updateMediaFor(post: AbstractPost, success: @escaping (() -> Void)) {
+        let mediaToUpdate = Array(post.media).filter { media in
+            guard let postID = media.postID else { return false }
+            return postID.intValue <= 0
+        }
+
+        mediaToUpdate.forEach { media in
+            media.postID = post.postID
+        }
+
+        let mediaService = MediaService(managedObjectContext: self.managedObjectContext)
+        mediaService.updateMedia(mediaToUpdate, overallSuccess: {
+            ContextManager.sharedInstance().save(self.managedObjectContext)
+            success()
+        }) { error in
+            success()
```

Comment:
Can you provide some background regarding why on failure we are reporting success?

The reason I'm asking is because if we decide we want to mute a specific error, it's usually best to leave that decision for the code that's calling this method, rather than to have this method automatically silence the error.

Reply:
I deleted the comment that explained this when I refactored the method:

'// Sergio Estevao (2018-02-27): even if media fails to attach we are answering with success because the post uploaded successful and the only thing that failed was attaching the media to it.'
I updated the callback to `completion` instead of `success` as Eric suggested but realize that you are right and it would be better handled by the caller so I added a failure block, logging the failure and added back the comment.